mindmeld/auto_annotator.py

```diff
@@ -362,32 +353,29 @@ def _resolve_conflicts(self, current_entities, annotated_entities, config):
         return base_entities
 
     # pylint: disable=R0201
-    def _unannotate_query(self, processed_query, remove_entities, config):
-        """ Removes specified entities in a processed query. If all entities are being
+    def _unannotate_query(self, processed_query, remove_entities):
+        """Removes specified entities in a processed query. If all entities are being
         removed, this function will not remove entities that the annotator does not support
         unless it is explicitly specified to do so in the config with the param
-        "unannotate_supported_entities_only" (boolean).
+        "unannotate_supported_entities_only" (bool).
 
         Args:
             processed_query (ProcessedQuery): A processed query.
             remove_entities (list): List of entities to remove.
-            config (dict): Config to use instead of the class config.
         """
-        config = config or self.config
-        remove_supported_only = config["unannotate_supported_entities_only"]
         keep_entities = []
         for query_entity in processed_query.entities:
             if remove_entities == ["*"]:
                 is_supported_entity = self.valid_entity_check(query_entity.entity.type)
-                if remove_supported_only and not is_supported_entity:
+                if self.unannotate_supported_entities_only and not is_supported_entity:
```

Comment:
`self.unannotate_supported_entities_only and not is_supported_entity and keep_entities.append(query_entity)`